### PR TITLE
[SYCL][NATIVECPU] Emit subhandler as LLVM IR

### PR DESCRIPTION
**Pull request by PietroGhg**
This PR allows the emit the Native CPU subhandler as LLVM-IR, instead of emitting its declaration as C++ code in the Native CPU integration header. This is a first step towards removing the Native CPU integration header, using clang-offload-wrapper and also supporting multiple SYCL targets alongside SYCL Native CPU, which we will implement in following PRs.

### CODE REVIEW

File: clang/test/CodeGenSYCL/native_cpu_basic.cpp [hunk 1]
```diff
@@ -51,10 +51,8 @@ void gen() {
 }
 
 // Check name mangling and kernel_arg_type metadata
```
Comment (elizabethandrews) [inline clang/test/CodeGenSYCL/native_cpu_basic.cpp:53]: This comment needs to be updated to remove mention of metadata since we no longer generate it

**Review by elizabethandrews — APPROVED** [review]
FE changes LGTM other than a minor change to update comment

File: clang/test/CodeGenSYCL/native_cpu_basic.cpp (hunk 1 above)
Comment (PietroGhg) [reply clang/test/CodeGenSYCL/native_cpu_basic.cpp]: I've update the comments in the test, thanks for spotting this

**Review by uwedolinsky — APPROVED** [review]

**Comment by PietroGhg** [comment]
@intel/llvm-gatekeepers hello, I think that this PR is ready to be merged :)

**Comment by dm-vodopyanov** [comment]
Failure in post-commit:
```
Failed Tests (1):
  SYCL :: check_device_code/native_cpu/native_cpu_subhandler.cpp
```
@PietroGhg could you please take a look asap?

**Comment by PietroGhg** [comment]
> Failure in post-commit:
> 
> ```
> Failed Tests (1):
>   SYCL :: check_device_code/native_cpu/native_cpu_subhandler.cpp
> ```
> 
> @PietroGhg could you please take a look asap?

I think that the failure is due to the fact the IR printer is not printing out the basic block name, https://github.com/intel/llvm/pull/10350 should fix it

**Pull request merged**